File: testers/combined/java/generated-and-checked-in/avro_events/combined/avro_events/consumer/TreeNodeHandler.java
package combined.avro_events.consumer;

import com.example.events.TreeNode;
import combined.avro_events.header.StandardHeaders;
import java.lang.Void;
import java.util.concurrent.CompletableFuture;

/** Handler interface for tree-node topic events */
public interface TreeNodeHandler {
  /** Handle a message from the topic */
  CompletableFuture<Void> handle(
    String key,
    TreeNode value,
    StandardHeaders headers
  );
}

File: testers/combined/kotlin/generated-and-checked-in/avro_events/combined/avro_events/consumer/TreeNodeHandler.kt
package combined.avro_events.consumer

import com.example.events.TreeNode
import combined.avro_events.header.StandardHeaders
import io.smallrye.mutiny.Uni

/** Handler interface for tree-node topic events */
interface TreeNodeHandler {
  /** Handle a message from the topic */
  abstract fun handle(
    key: kotlin.String,
    value: TreeNode,
    headers: StandardHeaders
  ): Uni<Unit>
}